### PR TITLE
Update `config` and run Detekt during the build

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR does the following:

1. Updates to the latest `config`.
2. Utilizes `Spine` dependency object.
3. Applies Detekt and Gradle Doctor to the module.
4. Adds in-place `pull` script.

Please note, this repository doesn't have tests. Its build workflows have been updated **NOT** to fail due to missing tests. 

The library version is bumped to `2.0.0-SNAPSHOT.5`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -62,42 +62,52 @@ all modules and does not describe the project structure per-subproject.
...
     <artifactId>error_prone_annotations</artifactId>
-    <version>2.13.1</version>
+    <version>2.16</version>
     <scope>provided</scope>
   </dependency>
   <dependency>
     <groupId>com.google.errorprone</groupId>
     <artifactId>error_prone_core</artifactId>
-    <version>2.13.1</version>
+    <version>2.16</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:89]: ignore

File: pom.xml [hunk 2]
```diff
@@ -62,42 +62,52 @@ all modules and does not describe the project structure per-subproject.
...
   <dependency>
     <groupId>com.puppycrawl.tools</groupId>
     <artifactId>checkstyle</artifactId>
-    <version>10.1</version>
+    <version>10.3.4</version>
+  </dependency>
+  <dependency>
+    <groupId>io.gitlab.arturbosch.detekt</groupId>
+    <artifactId>detekt-cli</artifactId>
+    <version>1.21.0</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:110]: ignore

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov @armiol Please take a look.

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM

**Pull request merged**